Core/GameEngine/Source/GameNetwork/NetCommandWrapperList.cpp

```diff
@@ -84,22 +84,42 @@ void NetCommandWrapperListNode::copyChunkData(NetWrapperCommandMsg *msg) {
 		return;
 	}
 
-	DEBUG_ASSERTCRASH(msg->getChunkNumber() < m_numChunks, ("MunkeeChunk %d of %d",
-		msg->getChunkNumber(), m_numChunks));
-	if (msg->getChunkNumber() >= m_numChunks)
+	UnsignedInt chunkNumber = msg->getChunkNumber();
+
+	if (chunkNumber >= m_numChunks) {
+		DEBUG_CRASH(("Data chunk number exceeds the number of chunks expected; chunk %d of %d", chunkNumber, m_numChunks));
 		return;
+	}
 
-	DEBUG_LOG(("NetCommandWrapperListNode::copyChunkData() - copying chunk %d",
-		msg->getChunkNumber()));
+	// we already received this chunk, no need to recopy it.
+	if (m_chunksPresent[chunkNumber] == TRUE) {
+		return;
+	}
 
-	if (m_chunksPresent[msg->getChunkNumber()] == TRUE) {
-		// we already received this chunk, no need to recopy it.
+	UnsignedInt chunkDataOffset = msg->getDataOffset();
+	UnsignedInt chunkDataLength = msg->getDataLength();
+
+	// TheSuperHackers @bugfix Mauller 04/12/2025 Prevent out of bounds memory access
+	if (chunkDataOffset >= m_totalDataLength) {
+		DEBUG_CRASH(("Data chunk offset beyond data size"));
```

Comment:
You can also massage the numbers in these new debug messages:

"Data chunk offset %d exceeds the total data lenght %d"

Reply:
Btw these are unsigned integers so use %u